pkg/reconciler/instances/istio/actions/performer.go

```diff
@@ -190,26 +179,46 @@ func (c *DefaultIstioPerformer) Install(kubeConfig, istioChart, version string,
 }
 
 func (c *DefaultIstioPerformer) PatchMutatingWebhook(context context.Context, kubeClient kubernetes.Client, logger *zap.SugaredLogger) error {
-	patchContent := []webhookPatchJSON{{
-		Op:   "add",
-		Path: "/webhooks/4/namespaceSelector/matchExpressions/-",
-		Value: webhookPatchJSONValue{
-			Key:      "gardener.cloud/purpose",
-			Operator: "NotIn",
-			Values: []string{
-				"kube-system",
-			},
-		},
-	}}
-
-	patchContentJSON, err := json.Marshal(patchContent)
+	clientSet, err := kubeClient.Clientset()
 	if err != nil {
 		return err
 	}
 
-	logger.Info("Patching istio-sidecar-injector MutatingWebhookConfiguration...")
+	const primary = "istio-revision-tag-default"
+	const secondary = "istio-sidecar-injector"
+	candidatesNames := []string{primary, secondary}
 
-	err = kubeClient.PatchUsingStrategy(context, "MutatingWebhookConfiguration", "istio-sidecar-injector", "istio-system", patchContentJSON, types.JSONPatchType)
+	webhookNameToChange := "auto.sidecar-injector.istio.io"
+
+	requiredLabelSelector := metav1.LabelSelectorRequirement{
+		Key:      "gardener.cloud/purpose",
+		Operator: "NotIn",
+		Values:   []string{"kube-system"},
+	}
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		whConf, err := c.selectWebhookConfFormCandidates(context, candidatesNames, clientSet)
+		if err != nil {
+			return err
+		}
+		var whFound bool
+		for i := range whConf.Webhooks {
+			if whConf.Webhooks[i].Name == webhookNameToChange {
+				whFound = true
+				whConf.Webhooks[i].NamespaceSelector.MatchExpressions = append(
+					whConf.Webhooks[i].NamespaceSelector.MatchExpressions,
+					requiredLabelSelector,
+				)
+			}
+		}
+		if !whFound {
+			return fmt.Errorf("could not find webhook %s in WebhookConfiguration %s", webhookNameToChange, whConf.Name)
+		}
+		_, err = clientSet.AdmissionregistrationV1().
```

Comment:
IMO we should align code formatting - either like this or like with other one-liners such as: `clientSet.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context, webhookName, metav1.GetOptions{})`